File: Chanseok/프로그래머스/Lv2/42578. 위장/위장.java
import java.util.HashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;
import java.util.Map.Entry;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;

        Map<String, Integer> map = new HashMap<>();
        for(int i = 0; i < clothes.length; ++i) {
            map.put(clothes[i][1], map.getOrDefault(clothes[i][1], 0)+1);
        }

        for(String key : map.keySet()) {
            answer *= (map.get(key)+1);
        }
        --answer;

        return answer;
    }
}

File: Seonguk/프로그래머스/위장/위장.py
from collections import defaultdict

def solution(clothes):
    dic=defaultdict(int)
    # 각 의상마다 몇개있는지 저장
    for i in range(len(clothes)):
        dic[clothes[i][1]]+=1
    
    answer=1
    for val in dic.values():
        answer*=(val+1) # 아무것도 안입는 경우의 수 + 1
    
    return answer-1 # 최소 하나의 의상을 입어야하므로 - 1
